### ISSUE TITLE
Stop program on figure close

### ISSUE THREAD

**Issue by wangwuqi** [stsievert/python-drawnow#9]
I put drawnow in the while loop, it can get the right plot,but when I try to close the figure, it regenerate a new plot automatically and continue plotting.Then how shoulde I do?

**Comment by stsievert** [comment]
drawnow should only refresh the figure, not create a new one (unless the correct keywords args are changed I think).

Is drawnow initialized correctly? Is the call is the correct place? Can you provide a code snippet and a minimal working example?

**Comment by wangwuqi** [comment]
I try to plot the streaming data from A NI device, the sampling rate is **1000Hz**,  can you offer help and suggestions? 

``` python
from pylab import *
from drawnow import drawnow, figure

import nidaqmx
from nidaqmx.stream_readers import AnalogSingleChannelReader,AnalogMultiChannelReader
import numpy as np
import time,datetime

plt.ion()

def draw_fig():
#    plt.ylim(-1,1)
    line, = plt.plot(data, 'b-',linewidth=0.6)

data = np.zeros([200])

while 1:
    with nidaqmx.Task() as read_task:
        read_task.ai_channels.add_ai_voltage_chan('Dev1/ai1', max_val=10,min_val=-10)
        reader = AnalogSingleChannelReader(read_task.in_stream)
        reader.read_many_sample(data,len(data))
        drawnow(draw_fig)
        plt.pause(0.0005)
```

**Comment by stsievert** [comment]
That's almost the same as the example from the readme except you're missing a call to `figure`. Could you try that?

**Comment by wangwuqi** [comment]
what does it mean by missing a call to figure?

**Comment by stsievert** [comment]
Like "the script does not have a function call to `figure` and I think it should". Put `figure()` before the while loop.

**Comment by wangwuqi** [comment]
still the same, I close the figure window and it regenerate a new figure window, and the figure window is always on the top front of other windows. the only way to stop plotting is to close the shell or press Ctrl + C to kill the program...
![image](https://user-images.githubusercontent.com/22237278/38092704-946c6da6-339b-11e8-9efc-fc64a6926585.png)

**Comment by stsievert** [comment]
Ah, I think I see the issue. Is it that you want the figure to refresh for a while then when you close the figure window to break out of the loop?

**Comment by wangwuqi** [comment]
I just want to close the figure window

**Comment by stsievert** [comment]
I do present an option to exit the program completely. If you do `drawnow(..., confirm=True)` every time it refreshes the figure it will ask if you want to exit.

Is that sufficient?

**Comment by wangwuqi** [comment]
```
Hit <Enter> to continue ("d" for debugger and "x" to exit): 
```
and the plot stay still, then I hit 'Enter', the curve moves a little, this is not what I want

**Comment by stsievert** [comment]
I think you can pass `show_once=True` to `drawnow` to get what you want. That will call `plt.show` instead of `draw`, so it'll behave like a normal matplotlib figure.

>  this is not what I want

Can you express clearly in one or two sentences what you want?

**Comment by wangwuqi** [comment]
just want to close the figure window

**Comment by stsievert** [comment]
Here's a couple short descriptions of wants:

> 1. I have a use case where I need to see the plot each iteration. However, I need to close it every time it refreshes (because of ...), but still want it to refresh every time `drawnow` is called. How would I do this?
> 2. I've used drawnow, but am still struggling with something. When I close the figure window I'd like my entire program to quit. Is that possible?
> 3. I'm using `drawnow(..., confirm=True)`. It presents three options: continue, exit the program or enter the debugger. I'd like an option close the figure window too. Could you add that?

Are any of these accurate?

**Comment by wangwuqi** [comment]
description 2 is close to my want, close the figure but not the python shell

**Comment by stsievert** [comment]
Rephrased:

> I have `drawnow` in an infinite while loop. I have a use case where I want to watch the figure over time. When I'm satisfied, I'd like to close the figure window and have my program to stop execution. Currently, it opens a new figure window and does not stop execution.

Is this more accurate?

**Comment by wangwuqi** [comment]
Yes

发自我的手机

**Comment by wangwuqi** [comment]
I record the scene, you can take a look at it
![record](https://user-images.githubusercontent.com/22237278/38123909-9333b17c-3410-11e8-885e-b9a2602d6cb4.gif)

**Comment by stsievert** [comment]
Yup, I understand now. I'm guessing I'll do something like

``` python
def drawnow(..., stop_on_close=False):
    if stop_on_close and not plt.get_fignums():
        sys.exit()
```

I need to make sure that `drawnow` depends on a `figure` being called

**Comment by stsievert** [comment]
Closed by 6e9c8fba4fb7f1e0e89460147e93ba5b9b0c0ba3
